### PR TITLE
Add/update AWS Local Zone workflow

### PR DESCRIPTION
**Pull request by yunjiang29**
**Added new workflow to cover more scenarios**
* `cucushift-installer-rehearse-aws-ipi-network-mtu-localzone-sdn`
* `cucushift-installer-rehearse-aws-ipi-disconnected-private-network-mtu-localzone`

**New Local Zone profiles**
* 4.12 x86 stable & nightly
  * aws-ipi-disconnected-network-mtu-localzone-p2-f14
  * aws-ipi-disconnected-network-mtu-localzone-p3-f28-destructive
  * aws-ipi-network-mtu-localzone-sdn-p2-f7
  * aws-ipi-network-mtu-localzone-sdn-p3-f28-destructive

/hold

### CODE REVIEW

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-aws-ipi-disconnected-network-mtu-localzone-p2-f14

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-stable-aws-ipi-network-mtu-localzone-p2-f7

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-stable-aws-ipi-network-mtu-localzone-sdn-p2-f7

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-aws-ipi-disconnected-network-mtu-localzone-p2-f14

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-stable-aws-ipi-network-mtu-localzone-sdn-p2-f7

**Comment by jianlinliu** [comment]
Is there different test points between `aws-ipi-disconnected-network-mtu-localzone` and `aws-ipi-network-mtu-localzone` ?

**Comment by yunjiang29** [comment]
> Is there different test points between `aws-ipi-disconnected-network-mtu-localzone` and `aws-ipi-network-mtu-localzone` ?

Lower MTU may impact pulling images as well, so added disconnected profile for Local Zone

**Comment by yunjiang29** [comment]
All installs get passed.

**Comment by jianlinliu** [comment]
shall we also take arm into condieration with lower priority, e.g: f28?

**Comment by yunjiang29** [comment]
LZ doesn't support arm instances and LZ Phase 1&2 will be released in 4.13, so it's better to keep it in a _higher_ frequency, WDYT?

**Comment by jianlinliu** [comment]
> LZ doesn't support arm instances

Oh, really? I saw `aws-ipi-network-mtu-localzone` was already added in arm nightly jobs, and I checked the job history, most of them get passed.

If we confirm local zone testing is only supported in amd ci jobs, it is fine to set a higher test frequency.

**Comment by yunjiang29** [comment]
> Oh, really? I saw aws-ipi-network-mtu-localzone was already added in arm nightly jobs, and I checked the job history, most of them got passed.

A mistake (no ARM instance type available in LZ) + an automation bug (we didn't check the number of nodes):

see the following job, it's not possible to create a worker node on LZ:
https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/logs/periodic-ci-openshift-openshift-tests-private-release-4.12-arm64-stable-aws-ipi-network-mtu-localzone-p2-f14/1620746734219038720/artifacts/aws-ipi-network-mtu-localzone-p2-f14/ipi-install-install-aws/artifacts/manifest_localzone_machineset.yaml

```
      providerSpec:
        value:
          ami:
            id: ami-02a574449d4f4d280. <- ARM
...
          instanceType: c5d.2xlarge     <- x86
```

install succeeded, but no worker node on LZ:
https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/logs/periodic-ci-openshift-openshift-tests-private-release-4.12-arm64-stable-aws-ipi-network-mtu-localzone-p2-f14/1620746734219038720/artifacts/aws-ipi-network-mtu-localzone-p2-f14/gather-extra/artifacts/oc_cmds/nodes

```
NAME                          STATUS   ROLES                  AGE   VERSION           INTERNAL-IP   EXTERNAL-IP   OS-IMAGE                                                        KERNEL-VERSION                  CONTAINER-RUNTIME
ip-10-0-53-203.ec2.internal   Ready    control-plane,master   8h    v1.25.4+a34b9e9   10.0.53.203   <none>        Red Hat Enterprise Linux CoreOS 412.86.202301191053-0 (Ootpa)   4.18.0-372.40.1.el8_6.aarch64   cri-o://1.25.2-4.rhaos4.12.git66af2f6.el8
ip-10-0-63-133.ec2.internal   Ready    worker                 8h    v1.25.4+a34b9e9   10.0.63.133   <none>        Red Hat Enterprise Linux CoreOS 412.86.202301191053-0 (Ootpa)   4.18.0-372.40.1.el8_6.aarch64   cri-o://1.25.2-4.rhaos4.12.git66af2f6.el8
ip-10-0-75-119.ec2.internal   Ready    control-plane,master   8h    v1.25.4+a34b9e9   10.0.75.119   <none>        Red Hat Enterprise Linux CoreOS 412.86.202301191053-0 (Ootpa)   4.18.0-372.40.1.el8_6.aarch64   cri-o://1.25.2-4.rhaos4.12.git66af2f6.el8
ip-10-0-78-40.ec2.internal    Ready    worker                 8h    v1.25.4+a34b9e9   10.0.78.40    <none>        Red Hat Enterprise Linux CoreOS 412.86.202301191053-0 (Ootpa)   4.18.0-372.40.1.el8_6.aarch64   cri-o://1.25.2-4.rhaos4.12.git66af2f6.el8
ip-10-0-82-58.ec2.internal    Ready    control-plane,master   8h    v1.25.4+a34b9e9   10.0.82.58    <none>        Red Hat Enterprise Linux CoreOS 412.86.202301191053-0 (Ootpa)   4.18.0-372.40.1.el8_6.aarch64   cri-o://1.25.2-4.rhaos4.12.git66af2f6.el8
ip-10-0-95-209.ec2.internal   Ready    worker                 8h    v1.25.4+a34b9e9   10.0.95.209   <none>        Red Hat Enterprise Linux CoreOS 412.86.202301191053-0 (Ootpa)   4.18.0-372.40.1.el8_6.aarch64   cri-o://1.25.2-4.rhaos4.12.git66af2f6.el8
```

Will fix above two issues, thanks @jianlinliu

**Comment by yunjiang29** [comment]
Nodes created by LZ profile should be like this:

https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/logs/periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-aws-ipi-network-mtu-localzone-p2-f14/1618618310625071104/artifacts/aws-ipi-network-mtu-localzone-p2-f14/gather-extra/artifacts/oc_cmds/nodes
```
NAME                           STATUS   ROLES                  AGE   VERSION           INTERNAL-IP    EXTERNAL-IP   OS-IMAGE                                                        KERNEL-VERSION                 CONTAINER-RUNTIME
ip-10-0-129-206.ec2.internal   Ready    edge,worker,wscan      9h    v1.25.4+a34b9e9   10.0.129.206   <none>        Red Hat Enterprise Linux CoreOS 412.86.202301191053-0 (Ootpa)   4.18.0-372.40.1.el8_6.x86_64   cri-o://1.25.2-4.rhaos4.12.git66af2f6.el8
ip-10-0-55-185.ec2.internal    Ready    worker,wscan           9h    v1.25.4+a34b9e9   10.0.55.185    <none>        Red Hat Enterprise Linux CoreOS 412.86.202301191053-0 (Ootpa)   4.18.0-372.40.1.el8_6.x86_64   cri-o://1.25.2-4.rhaos4.12.git66af2f6.el8
ip-10-0-59-181.ec2.internal    Ready    control-plane,master   10h   v1.25.4+a34b9e9   10.0.59.181    <none>        Red Hat Enterprise Linux CoreOS 412.86.202301191053-0 (Ootpa)   4.18.0-372.40.1.el8_6.x86_64   cri-o://1.25.2-4.rhaos4.12.git66af2f6.el8
ip-10-0-73-39.ec2.internal     Ready    control-plane,master   10h   v1.25.4+a34b9e9   10.0.73.39     <none>        Red Hat Enterprise Linux CoreOS 412.86.202301191053-0 (Ootpa)   4.18.0-372.40.1.el8_6.x86_64   cri-o://1.25.2-4.rhaos4.12.git66af2f6.el8
ip-10-0-77-200.ec2.internal    Ready    worker,wscan           9h    v1.25.4+a34b9e9   10.0.77.200    <none>        Red Hat Enterprise Linux CoreOS 412.86.202301191053-0 (Ootpa)   4.18.0-372.40.1.el8_6.x86_64   cri-o://1.25.2-4.rhaos4.12.git66af2f6.el8
ip-10-0-80-178.ec2.internal    Ready    worker,wscan           9h    v1.25.4+a34b9e9   10.0.80.178    <none>        Red Hat Enterprise Linux CoreOS 412.86.202301191053-0 (Ootpa)   4.18.0-372.40.1.el8_6.x86_64   cri-o://1.25.2-4.rhaos4.12.git66af2f6.el8
ip-10-0-80-186.ec2.internal    Ready    control-plane,master   10h   v1.25.4+a34b9e9   10.0.80.186    <none>        Red Hat Enterprise Linux CoreOS 412.86.202301191053-0 (Ootpa)   4.18.0-372.40.1.el8_6.x86_64   cri-o://1.25.2-4.rhaos4.12.git66af2f6.el8
```

**Comment by jianlinliu** [comment]
Thanks for checking, are we going to remove localzone testing from arm jobs in this PR?

**Comment by yunjiang29** [comment]
@jianlinliu I will create another PRs to fix above issues.

This PR is ready for merge, thanks.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly.yaml [hunk 1]
```diff
@@ -148,6 +148,32 @@ tests:
     test:
     - chain: openshift-e2e-test-qe-destructive
     workflow: cucushift-installer-rehearse-aws-ipi-disconnected-private-cco-manual-security-token-service
+- as: aws-ipi-disconnected-network-mtu-localzone-p2-f14
```
Comment (liangxia) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly.yaml:151]: The profile is `aws-ipi-disconnected-private-network-mtu-localzone`. I think we are supposed to reflect the profile name in test name

Comment (yunjiang29) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly.yaml:151]: Avoid name length restriction, like test name `aws-ipi-disconnected-cco-manual-sts-p3-f28-destructive` -> `aws-ipi-disconnected-private-cco-manual-security-token-service` in https://github.com/openshift/release/blob/master/ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly.yaml#L138-L150 

And in Prow CI, all _disconnected_ clusters are _private_, so there is no ambiguity.

**Pull request merged**